### PR TITLE
Improve pom and project structure

### PR DESCRIPTION
**Pull request by artem-v**
+ testlib
+ benchmarks (install IDEA plugin!)
+ consmetical updates in poms

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -117,7 +175,7 @@
                     <plugin>
                         <groupId>org.apache.maven.plugins</groupId>
                         <artifactId>maven-gpg-plugin</artifactId>
-                        <version>1.5</version>
+                        <version>1.4</version>
```
Comment (antonkharenko) [inline pom.xml]: why 1.4 ?

Comment (artem-v) [inline pom.xml]: IDEA were complaining on 1.5

**Comment by myroslavlisniak** [comment]
Rebase your commits on top of master and squash them

File: cluster/pom.xml [hunk 2]
```diff
@@ -1,34 +1,25 @@
...
-	<artifactId>servicefabric-cluster</artifactId>
-    <packaging>jar</packaging>
-	
-	<name>ServiceFabric/Cluster</name>
-	
-	<properties>
-		<jmock.version>2.6.0</jmock.version>
-	</properties>
-	
+    <artifactId>cluster</artifactId>
```
Comment (antonkharenko) [inline cluster/pom.xml]: Lets not rename artifactId

File: pom.xml [hunk 3]
```diff
@@ -1,6 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
 
     <modelVersion>4.0.0</modelVersion>
+    <parent>
```
Comment (antonkharenko) [inline pom.xml]: No need for sonatype parent

File: pom.xml [hunk 4]
```diff
@@ -48,33 +59,71 @@
...
+            </dependency>
+            <dependency>
+                <groupId>io.reactivex</groupId>
+                <artifactId>rxjava</artifactId>
+                <version>${rxjava.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>com.google.code.findbugs</groupId>
+                <artifactId>jsr305</artifactId>
+                <version>${findbugs.version}</version>
```
Comment (antonkharenko) [inline pom.xml]: Lets put here to dependency management all dependencies we use in project and in each module their dependencies (without versions)

File: pom.xml [hunk 5]
```diff
@@ -48,33 +59,71 @@
...
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
 
     <build>
+        <sourceDirectory>src</sourceDirectory>
+        <testSourceDirectory>test</testSourceDirectory>
```
Comment (antonkharenko) [inline pom.xml]: I prefere to stay with default project structure

File: pom.xml [hunk 6]
```diff
@@ -48,31 +59,75 @@
...
+                <version>${protostuff.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>com.google.protobuf</groupId>
+                <artifactId>protobuf-java</artifactId>
+                <version>${protobuf.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>io.netty</groupId>
+                <artifactId>netty-all</artifactId>
```
Comment (antonkharenko) [inline pom.xml]: Lets not use netty all, but specific modules

File: cluster/pom.xml [hunk 7]
```diff
@@ -1,34 +1,25 @@
...
-		<jmock.version>2.6.0</jmock.version>
-	</properties>
-	
+    <artifactId>servicefabric-cluster</artifactId>
+    <name>ServiceFabric/Cluster</name>
     <dependencies>
         <dependency>
             <groupId>${project.groupId}</groupId>
-            <artifactId>servicefabric-transport</artifactId>
+            <artifactId>transport</artifactId>
```
Comment (antonkharenko) [inline cluster/pom.xml]: wrong artifactId

File: cluster/pom.xml [hunk 8]
```diff
@@ -1,34 +1,25 @@
...
             <version>${project.version}</version>
         </dependency>
         <dependency>
-            <groupId>org.jmock</groupId>
-            <artifactId>jmock-junit4</artifactId>
-			<version>${jmock.version}</version>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>testlib</artifactId>
```
Comment (antonkharenko) [inline cluster/pom.xml]: also

File: transport/pom.xml [hunk 9]
```diff
@@ -1,74 +1,72 @@
...
-		<findbugs.version>2.0.3</findbugs.version>
-		<protobuf.version>2.4.1</protobuf.version>
-		<rxjava.version>1.0.6</rxjava.version>
-	</properties>
-	
+    <name>ServiceFabric/Transport</name>
     <dependencies>
         <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>testlib</artifactId>
```
Comment (antonkharenko) [inline transport/pom.xml]: artifactId

**Comment by antonkharenko** [comment]
+1

**Pull request merged**